File: 0141-linked-list-cycle/0141-linked-list-cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head, history = []) {
    if(!head){return false}
    if(history.includes(head)){return true}
    history.push(head);
    return hasCycle(head.next, history);
};

File: 0141-linked-list-cycle/0141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None
"""
myDict = {myKey: myValue}
myObject.myKey = myValue

head = {
    val: 1,
    next: {
        val: 2,
        next: None
    }
}
"""

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        seen = []
        
        curr = head
        while curr != None:
            if curr in seen:
                return True
            seen.append(curr)
            curr = curr.next
        return False